### DISCUSSION TITLE
Can I use multiple overlapping @@unique tags?

### DISCUSSION THREAD

**Discussion by StokleyO** [prisma/prisma#25620] [Q&A] [answered]
### Question

Hey all, I think I know the answer to this, but am currently in the process of developing a multi-month project and as such, won't be able to easily test my assumption for a while. I have a Prisma model, using Postgresql, that has a model that functions similar to Tags. On this model, I have two @@unique attributes and Prisma is not throwing an error. However, if I understand correctly, it should, because the first @@unique tag can never be unique. Let me know your thoughts
```prisma
model Tags {
    id                String               @id @default(uuid())
    text              String
    type              TagType
    numberOfVideos          Int?

    @@unique([text, type])
    @@unique([text, type, duration])
}
```
### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @StokleyO 👋 

Prisma does not throw an error when you define multiple `@@unique` constraints in your model. The schema definition you provided is valid Prisma syntax.

The first constraint `@@unique([text, type])` is a subset of the second constraint `@@unique([text, type, duration])`. This means that the second constraint is more restrictive than the first one. While this might seem redundant, it's not inherently wrong or problematic from a database perspective.

The first constraint ensures that the combination of text and type is unique across all records. The second constraint ensures that the combination of text, type, and duration is unique across all records.

**Reply by StokleyO** [reply]
Thank you for verifying this! I was able to find another use case around what I was trying to solve but it's helpful to know how these constraints work with one another
